libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotification.kt

```diff
@@ -92,22 +92,24 @@ internal class MapboxNavigationNotification : NavigationNotification {
     }
 
     // Package private (no modifier) for testing purposes
-    fun generateArrivalTime(routeProgress: RouteProgress, time: Calendar): String? =
-        mapboxNavigation.let { mapboxNavigation ->
-            val options = mapboxNavigation.options()
-            val legDurationRemaining = routeProgress.currentLegProgress().durationRemaining()
-            val timeFormatType = options.timeFormatType()
-            val arrivalTime =
-                formatTime(time, legDurationRemaining, timeFormatType, isTwentyFourHourFormat)
-            String.format(etaFormat, arrivalTime)
-        }
+    fun generateArrivalTime(routeProgress: RouteProgress, time: java.util.Calendar): String? =
+            ifNonNull(mapboxNavigation) { mapboxNavigation ->
+                val options = mapboxNavigation.options()
+                val legDurationRemaining = routeProgress.currentLegProgress().durationRemaining()
+                val timeFormatType = options.timeFormatType()
+                ifNonNull(legDurationRemaining) { durationRemaining ->
+                    val arrivalTime =
+                            formatTime(time, durationRemaining, timeFormatType, isTwentyFourHourFormat)
+                    String.format(etaFormat, arrivalTime)
+                }
+            }
```

Comment:
WDYT? Does it looks better or not?
```suggestion
            ifNonNull(
                mapboxNavigation?.options(), 
                routeProgress.currentLegProgress().durationRemaining()
            ) { options, legDurationRemaining  ->
                val arrivalTime = formatTime(
                    time, 
                    legDurationRemaining, 
                    options.timeFormatType(), 
                    isTwentyFourHourFormat
                )
                String.format(etaFormat, arrivalTime)
            }
```